ArduPlane/mode.cpp

```diff
@@ -0,0 +1,85 @@
+#include "mode.h"
+#include "Plane.h"
+
+Mode::Mode()
+{
+}
+
+void Mode::exit()
+{
+    // call sub-classes exit
+    _exit();
+}
+
+bool Mode::enter()
+{
+    // cancel inverted flight
+     plane.auto_state.inverted_flight = false;
```

Comment:
You are nuking vast amounts of state here.  If the subclass enter fails - what happens?  I *believe* Plane doesn't let that happen - but this is a landmine waiting to happen.

Reply:
yes, it has always been a landmine. If we change things to allow set_mode to ever fail then a whole redesign will be needed.